app/src/main/java/com/example/tasky/login/components/TaskyTextField.kt

```diff
@@ -0,0 +1,109 @@
+package com.example.tasky.login.components
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import com.example.tasky.R
+import com.example.tasky.ui.theme.DarkGray
+import com.example.tasky.ui.theme.Gray
+import com.example.tasky.ui.theme.Gray2
+import com.example.tasky.ui.theme.Green
+import com.example.tasky.ui.theme.LightBlue2
+import com.example.tasky.ui.theme.LightGray2
+import com.example.tasky.ui.theme.Red
+
+@Composable
+fun TaskyTextField(
+    isPassword: Boolean = false,
+    isValid: Boolean = false,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    imeAction: ImeAction = ImeAction.Done,
+    onValueChange: (String) -> Unit
+) {
+    var value by remember { mutableStateOf(TextFieldValue()) }
```

Comment:
This value won't survive config changes, so the text field will be empty after a screen rotation for example. I'd keep the whole screen state part of a ViewModel to avoid that